### ISSUE TITLE
Math on aggregates in single query

### ISSUE THREAD

**Issue by ross-nordstrom** [influxdb/influxdb#1299]
**TL;DR:**  Without the ability to use subqueries or reuse aggregations in-line in my queries, I can't seem to do what I need to do: `SELECT SUM(sum) / SUM(count) as count FROM ...`

Is there a way to accomplish my goal that I'm not aware of? Will we have subqueries or the ability to reuse aggregations in-line in 0.9.0? Thoughts/advice?

## Scenario
My use case is trying to calculate a "mean of means" on non-uniform pre-aggregrated series. 

In my pre-aggregated series, I store things like `mean`, `sum`, `count`, etc. Note that by non-uniform, I mean each time-bucket would have a different number of points, so to re-aggregate I need a weighted mean. Because of this, I cannot get the mean with the current query language. Naively using `MEAN(mean)` is incorrect, so what I really need to do is `SUM(sum) / SUM(count) as mean`. This does not work for me.

By manually using a continuous query as a poor-man's subquery ([+1 for subqueries!](https://github.com/influxdb/influxdb/issues/52)), I can effectively get the correct mean, but that approach is ugly and I woudn't want to use it in production.

### Pre-aggregated series
```
select * from seriesA
```

time	| sequence_number | 	sum | 	count |	mean
---|---|---|---|---
1420738130469 |	512600001 |	200 |	7 |	28.571
1420738119145 |	512590001 |	80 |	5 |	16
1420738108619 |	512580001 |	10 |	2 |	5

### Re-aggregate it
What if I want to summarize the series with a single point? Or say I want to re-aggregate (such as with `1hour` buckets re-rolled to `1day` summaries).

#### Naive attempt (mean of means)
```
select SUM(sum), SUM(count) as count, MEAN(mean) from seriesA
```

time |	sum |	count |	mean
---|---|---|---
0 |	290 |	14 |	16.523666666666667

__But this is incorrect!__ Sum / count  should equal mean, but `290/14 = 20.7 !== 16.52` 

#### Weighted mean
So clearly we want to take `(SUM(sum) / SUM(count)) as mean`, but I can't construct a working query to this effect. In all cases I just get the `sum` and `count`, mean having been discarded.

Queries I've tried:
```
# Use the aggregations inline
select SUM(sum), SUM(count) as count, (SUM(sum) / SUM(count)) as mean from seriesA

# Maybe the result of the aggregation are available?
select SUM(sum) as mySum, SUM(count) as myCount, (mySum / myCount) as mean from seriesA
# ==> Error: Field "mySum" doesn't exist in series seriesA
```

#### Hacked subquery using continous query
Temporarily store the SUM(sum) and SUM(count). This is virtually a subquery!
```
select SUM(sum), SUM(count) as count from seriesA into reagg.seriesA
# ==> Continuous query created
```

**Finally! The result we need:**
```
select sum, count, (sum / count) as mean from reagg.seriesA
```

time |	sequence_number |	sum |	count |	mean
---|---|---|---|---
0 |	1 |	290 |	14 |	20.714285714285715

**Comment by davidblewett** [comment]
Duplicate of #438.
